<a href="https://colab.research.google.com/github/jakew4/project2/blob/main/coreproject2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# First Choice: NFL Combine Perfomance Data from 2009-2019

In [2]:
import pandas as pd

In [22]:
filename = '/content/NFL.csv'
df = pd.read_csv(filename)
df.info()
df.head(8)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3477 entries, 0 to 3476
Data columns (total 18 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   Year                 3477 non-null   int64  
 1   Player               3477 non-null   object 
 2   Age                  2927 non-null   float64
 3   School               3477 non-null   object 
 4   Height               3477 non-null   float64
 5   Weight               3477 non-null   float64
 6   Sprint_40yd          3303 non-null   float64
 7   Vertical_Jump        2780 non-null   float64
 8   Bench_Press_Reps     2572 non-null   float64
 9   Broad_Jump           2749 non-null   float64
 10  Agility_3cone        2260 non-null   float64
 11  Shuttle              2337 non-null   float64
 12  Drafted..tm.rnd.yr.  2254 non-null   object 
 13  BMI                  3477 non-null   float64
 14  Player_Type          3477 non-null   object 
 15  Position_Type        3477 non-null   o

Year                     Player   Age          School  Height      Weight  \
0  2009      Beanie Wells\WellCh00  20.0        Ohio St.  1.8542  106.594207   
1  2009        Will Davis\DaviWi99  22.0        Illinois  1.8796  118.387609   
2  2009    Herman Johnson\JohnHe23  24.0             LSU  2.0066  165.107623   
3  2009    Rashad Johnson\JohnRa98  23.0         Alabama  1.8034   92.079251   
4  2009        Cody Brown\BrowCo96  22.0     Connecticut  1.8796  110.676538   
5  2009   Trevor Canfield\CanfTr20  23.0      Cincinnati  1.9304  139.252858   
6  2009       Peria Jerry\JerrPe99  24.0     Mississippi  1.8796  135.624119   
7  2009  Garrett Reynolds\ReynGa20  21.0  North Carolina  2.0320  140.160042   

   Sprint_40yd  Vertical_Jump  Bench_Press_Reps  Broad_Jump  Agility_3cone  \
0         4.38          85.09              25.0      325.12            NaN   
1         4.84          83.82              27.0      292.10           7.38   
2         5.50            NaN              21.0         NaN            NaN   
3         4.49          93.98              15.0      304.80           7.09   
4         4.76          92.71              26.0      304.80           7.10   
5         5.28            NaN              29.0         NaN            NaN   
6         4.98            NaN               NaN         NaN            NaN   
7         5.32          55.88              19.0      238.76           7.87   

   Shuttle                          Drafted..tm.rnd.yr.        BMI  \
0      NaN   Arizona Cardinals / 1st / 31st pick / 2009  31.004194   
1     4.45  Arizona Cardinals / 6th / 204th pick / 2009  33.510073   
2      NaN  Arizona Cardinals / 5th / 167th pick / 2009  41.005821   
3     4.23   Arizona Cardinals / 3rd / 95th pick / 2009  28.312463   
4     4.40   Arizona Cardinals / 2nd / 63rd pick / 2009  31.327425   
5      NaN  Arizona Cardinals / 7th / 254th pick / 2009  37.368830   
6      NaN     Atlanta Falcons / 1st / 24th pick / 2009  38.388934   
7     4.88    Atlanta Falcons / 5th / 156th pick / 2009  33.945078   

  Player_Type      Position_Type Position Drafted  
0     offense    backs_receivers       RB     Yes  
1     defense  defensive_lineman       DE     Yes  
2     offense  offensive_lineman       OG     Yes  
3     defense     defensive_back       FS     Yes  
4     defense        line_backer      OLB     Yes  
5     offense  offensive_lineman       OG     Yes  
6     defense  defensive_lineman       DT     Yes  
7     offense  offensive_lineman       OT     Yes

In [12]:
df['Drafted'].value_counts()

Yes    2254
No     1223
Name: Drafted, dtype: int64

## About the dataset:
1. Source of data: https://www.kaggle.com/datasets/redlineracer/nfl-combine-performance-data-2009-2019 
  - data is from Kaggle
2. This dataset contains information from the NFL Combine (2009 to 2019), including the results from sports performance tests and draft outcomes.
3. The target could be a number of different columns/questions, most likely one of the following: 
  - Drafted? (Drafted column)
  - What position type or player type is most represented at the NFL Combine? (Position_Type or Position columns)
  - Do drafted and undrafted players perform differently on performance tests? 
4. Each row represents a player, his combine statistics, and his draft outcome. 
5. Classification problem
6. 16 features
7. 3477 rows
8. Draft outcomes (Draft round, and which pick of each round) are all in one column (Drafted..tm.rnd.yr.) and might need to be separated somehow. There are significant missing values in certain combine events, probably due to certain positions only participating in events relevant to their position. Imputing those values might be tricky. 


# Second Choice: Loan Prediction Problem Dataset

In [20]:
filepath = '/content/train_u6lujuX_CVtuZ9i.csv'
df2 = pd.read_csv(filepath)
df2.info()
df2.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 614 entries, 0 to 613
Data columns (total 13 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Loan_ID            614 non-null    object 
 1   Gender             601 non-null    object 
 2   Married            611 non-null    object 
 3   Dependents         599 non-null    object 
 4   Education          614 non-null    object 
 5   Self_Employed      582 non-null    object 
 6   ApplicantIncome    614 non-null    int64  
 7   CoapplicantIncome  614 non-null    float64
 8   LoanAmount         592 non-null    float64
 9   Loan_Amount_Term   600 non-null    float64
 10  Credit_History     564 non-null    float64
 11  Property_Area      614 non-null    object 
 12  Loan_Status        614 non-null    object 
dtypes: float64(4), int64(1), object(8)
memory usage: 62.5+ KB


Loan_ID Gender Married Dependents     Education Self_Employed  \
0  LP001002   Male      No          0      Graduate            No   
1  LP001003   Male     Yes          1      Graduate            No   
2  LP001005   Male     Yes          0      Graduate           Yes   
3  LP001006   Male     Yes          0  Not Graduate            No   
4  LP001008   Male      No          0      Graduate            No   

   ApplicantIncome  CoapplicantIncome  LoanAmount  Loan_Amount_Term  \
0             5849                0.0         NaN             360.0   
1             4583             1508.0       128.0             360.0   
2             3000                0.0        66.0             360.0   
3             2583             2358.0       120.0             360.0   
4             6000                0.0       141.0             360.0   

   Credit_History Property_Area Loan_Status  
0             1.0         Urban           Y  
1             1.0         Rural           N  
2             1.0         Urban           Y  
3             1.0         Urban           Y  
4             1.0         Urban           Y

## about the dataset: 
1. Source of data: https://www.kaggle.com/datasets/altruistdelhite04/loan-prediction-problem-dataset/code
  - the dataset is from kaggle
2. The dataset contains loan applicant info and the status of their loan
3. Target would most likely be Loan_Status.
4. One row represents a loan applicant and their loan status
5. This would most likely be a classification problem
6. 12 features 
7. 614 rows in the dataset 
8. Other than perhaps a relatively smaller number of features I don't anticipate any big challenges with this dataset. Missing values are few and can probably be easily imputed. 

In [21]:
df2['Loan_Status'].value_counts()

Y    422
N    192
Name: Loan_Status, dtype: int64